### PR TITLE
fix: TIME_GRAIN lookups for second/millisecond/unspecified grains

### PR DESCRIPTION
**Pull request by briangregoryholmes**
### Problem

  When viewing an infinite time range (`inf`) without a valid `smallest_time_grain`, the derived `activeTimeGrain` can be set to `TIME_GRAIN_SECOND`. Since `TIME_GRAIN_SECOND`, `TIME_GRAIN_MILLISECOND`, and `TIME_GRAIN_UNSPECIFIED` were missing from the `TIME_GRAIN` record, this caused runtime failures in UI code that performed lookups (e.g., formatting, pivot chip labels).

  ### Solution

  **Core fix:** Add missing grain entries to `TIME_GRAIN` record.

  **Additional cleanup:**
  - Replace redundant `FromURLParamTimeGrainMap` / `ToURLParamTimeGrainMapMap` with existing `DateTimeUnitToV1TimeGrain` / `V1TimeGrainToDateTimeUnit` from `new-grains.ts`
  - Use `effectiveGrain` (respects `minTimeGrain`) instead of raw `activeTimeGrain` in time series charts
  - Refactor `getAdjustedFetchTime` to use Luxon directly instead of legacy helper functions
  - Add `formatDateTimeByGrain` utility for grain-aware datetime formatting

  ### Changes

  #### Core fix
  > `config.ts`, `types.ts`

  Add `MILLISECOND`, `SECOND`, `UNSPECIFIED` to `TIME_GRAIN` record and `Period` enum

  #### Mapper consolidation
  > `mappers.ts`, `time-state.ts`, `TDDHeader.svelte`, `selectors.ts`, + 6 url-state files

  Remove duplicate mappers, use `V1TimeGrainToDateTimeUnit` throughout

  #### Time series
  > `MetricsTimeSeriesCharts.svelte`, `timeseries-data-store.ts`

  Use `effectiveGrain` with `isGrainAllowed` guard

  #### Utilities
  > `formatter.ts`, `ranges/index.ts`, `new-grains.ts`

  Add `formatDateTimeByGrain`, simplify `getAdjustedFetchTime`

  #### Tests
  > `formatter.spec.ts`

  Add 12 tests for `formatDateTimeByGrain`

  ---

  ### Technical debt

  This is a targeted fix. The underlying issue is type fragmentation across `TIME_GRAIN`, `V1TimeGrain`, `AvailableTimeGrain`, `Period`, and `TimeUnit`—all of which duplicate Luxon's `DateTimeUnit`. A TODO comment documents the path forward: consolidate around Luxon utilities per the pattern in `new-grains.ts`.

**Checklist:**
- [x] Covered by tests
- [x] Ran it and it works as intended
- [x] Reviewed the diff before requesting a review
- [x] Checked for unhandled edge cases
- [ ] Linked the issues it closes
- [x] Checked if the docs need to be updated. If so, create a separate Linear DOCS issue
- [x] Intend to cherry-pick into the release branch
- [x] I'm proud of this work!

### CODE REVIEW

File: web-common/src/lib/time/ranges/formatter.spec.ts [hunk 1]
```diff
@@ -370,3 +373,116 @@ describe("prettyFormatTimeRange", () => {
...
+    const result = formatDateTimeByGrain(testDate, V1TimeGrain.TIME_GRAIN_YEAR);
+    expect(result).toBe("2025");
+  });
+
+  it("formats quarter grain with month and year", () => {
+    const result = formatDateTimeByGrain(
+      testDate,
+      V1TimeGrain.TIME_GRAIN_QUARTER,
+    );
+    expect(result).toMatch(/Sep.*2025/);
```
Comment (AdityaHegde) [inline web-common/src/lib/time/ranges/formatter.spec.ts]: Whats the idea behind using `toMatch` and `not.toMatch` instead of `toEqual`? We are hardcoding the test date. So would be good to see how the format looks like in the tests.

File: web-common/src/lib/time/ranges/index.ts [hunk 2]
```diff
@@ -180,36 +181,28 @@ export function getAdjustedFetchTime(
...
-    zone,
-  );
+  const luxonUnit = V1TimeGrainToDateTimeUnit[interval];
 
-  const offsetedEndTime = getOffset(
-    endTime,
-    TIME_GRAIN[interval].duration,
-    TimeOffsetType.ADD,
-  );
+  // Should only fail if somehow the Luxon unit is invalid
```
Comment (AdityaHegde) [inline web-common/src/lib/time/ranges/index.ts:187]: Would be good to add some tests for this since it is core to our range option selection.

File: web-common/src/lib/time/ranges/formatter.spec.ts (hunk 1 above)
Comment (briangregoryholmes) [reply web-common/src/lib/time/ranges/formatter.spec.ts]: This was a holdover from thinking about covering more locale variations (where the parts are the same but may have different separators), but it's hardcoded as UTC for now so I'll switch to toBe

File: web-common/src/lib/time/ranges/index.ts (hunk 2 above)
Comment (briangregoryholmes) [reply web-common/src/lib/time/ranges/index.ts:187]: This function has test coverage already in web-common/src/lib/time/ranges/index.spec.ts, but I added ones specifically for undefined and invalid grain.
